docs/userguides/testing.md

```diff
@@ -217,3 +217,45 @@ def test_account_balance(project, owner, receiver, nft):
     expect = quantity
     assert actual == expect
 ```
+
+## Multi-chain Testing
+
+The Ape framework supports connecting to alternative providers in tests.
+The easiest way to achieve this is to use the `networks` provider context-manager.
+
+```python
+# Switch to Fantom mid test
+def test_my_fantom_test(networks):
+    # The test starts in 1 ecosystem but switches to another
+    assert networks.provider.network.ecosystem.name == "ethereum"
+    
+    with networks.fantom.local.use_provider("test") as provider:
+        assert provider.network.ecosystem.name == "fantom"
+    
+    # You can also use the context manager like this:
+    with networks.parse_network_choice("fantom:local:test") as provider:
+       assert provider.network.ecosystem.name == "fantom"
+```
+
+Use fixtures to assist as well:
+
+```python
+import pytest
+
+
+@pytest.fixture
+def stark_contract(networks, project):
```

Comment:
what happens if you use a provider with two different network contexts in a test?

Reply:
as long as the chain ID is different, it will establish a new connection and maintain that connection throughout the session.